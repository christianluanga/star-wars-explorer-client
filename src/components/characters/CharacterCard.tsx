import React from 'react'
import {Card, Button} from 'react-bootstrap'
import { Character } from '../types/types'
import '../characters/style.css';
import styled from 'styled-components';
//@ts-ignore
import { Link } from "@reach/router"

const CharacterCard = ({name, height, gender, mass, homeworld}: Character) => {
    return (
        <Card className="character-card" style={{ width: '19%', height: '25rem', marginLeft: '1%', marginBottom: '3rem'}}>
        <Card.Body className="mt-3">
            <Card.Title className="mb-5 text-center"><h3>{name}</h3></Card.Title>
            <Card.Text className="d-flex justify-content-center">
                <Item>Heigth: <Span>{height}</Span></Item>
            </Card.Text>
            <Card.Text className="d-flex justify-content-center">
                <Item>Mass: <Span>{mass}</Span></Item>
            </Card.Text>
            <Card.Text className="d-flex justify-content-center">
                <Item>Gender: <Span>{gender}</Span></Item>
            </Card.Text>
            <Card.Text className="d-flex justify-content-center">
                <Item>Homeworld: <Span>{homeworld}</Span></Item>
            </Card.Text>
        </Card.Body>
        <div className="d-flex justify-content-end mt-2">
        <Link to={`/character/${name}`}>
            <Button variant="danger">See More</Button>
        </Link>
        </div>
        </Card>
    )
}
const Item = styled.div`
    font-size: 20px;
`
const Span = styled.span`
    font-weight: bold;
`
export default CharacterCard
