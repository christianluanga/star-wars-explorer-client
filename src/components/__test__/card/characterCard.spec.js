import React from 'react'
import TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { screen , render ,act } from '@testing-library/react'
import CharacterCard from '../../characters/card/CharacterCard';
import { keyframes } from 'styled-components';

const mock = {
    name: "Luke Skywalker",
    mass: "77",
    height: "172",
    gender: "male",
    homeworld: "Tatooine"
}
  describe('renders without error', () => {
    test("The CharacterCard component renders correctly()", async()=>{
      const component = render(
        <CharacterCard  {...mock}/>
      );
      for(const [k,v] of Object.entries(mock)){
          expect(screen.getByTestId(`${mock.name}-${k}`)).toHaveTextContent(v)
      }
    })
    test("Get a snapshot of the CharacterCard component", ()=>{
      const tree = TestRenderer.create(<CharacterCard  {...mock}/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
  })
  });