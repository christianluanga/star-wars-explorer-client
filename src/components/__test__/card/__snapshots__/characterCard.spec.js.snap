// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without error Get a snapshot of the CharacterCard component 1`] = `
<div
  className="character-card mb-4 card"
  data-testid="Luke Skywalker"
>
  <div
    className="mb-5 text-center card-header"
  >
    <h3
      data-testid="Luke Skywalker-name"
    >
      Luke Skywalker
    </h3>
  </div>
  <div
    className="mt-3 card-body"
  >
    <p
      className="d-flex justify-content-center card-text"
    >
      <div
        className="sc-bdfBQB gYSCuo"
      >
        Heigth: 
        <span
          className="sc-gsTEea dZuxIm"
          data-testid="Luke Skywalker-height"
        >
          172
        </span>
      </div>
    </p>
    <p
      className="d-flex justify-content-center card-text"
    >
      <div
        className="sc-bdfBQB gYSCuo"
      >
        Mass: 
        <span
          className="sc-gsTEea dZuxIm"
          data-testid="Luke Skywalker-mass"
        >
          77
        </span>
      </div>
    </p>
    <p
      className="d-flex justify-content-center card-text"
    >
      <div
        className="sc-bdfBQB gYSCuo"
      >
        Gender: 
        <span
          className="sc-gsTEea dZuxIm"
          data-testid="Luke Skywalker-gender"
        >
          male
        </span>
      </div>
    </p>
    <p
      className="d-flex justify-content-center card-text"
    >
      <div
        className="sc-bdfBQB gYSCuo"
      >
        Homeworld: 
        <span
          className="sc-gsTEea dZuxIm"
          data-testid="Luke Skywalker-homeworld"
        >
          Tatooine
        </span>
      </div>
    </p>
  </div>
  <div
    className="d-flex justify-content-end mt-2"
  >
    <a
      href="/character/Luke Skywalker"
      onClick={[Function]}
    >
      <button
        className="btn btn-danger"
        disabled={false}
        type="button"
      >
        See More
      </button>
    </a>
  </div>
</div>
`;
